### ISSUE TITLE
Request signature verification always returns false

### ISSUE THREAD

**Issue by zafeirism** [AreYouFreeBusy/AlexaSkillsKit.NET#84]
### Context
I'm implementing a custom Alexa skill on a .net Web Api 2 project. It has very basic functionality so all of the request-response handling happens on my code and I only use AlexaSkillsKit.NET for signature verification:

```
private async Task<bool> IsValidAlexaSignature() {
    var signature = Request.Headers.Contains("Signature")
        ? Request.Headers.GetValues("Signature").FirstOrDefault()
        : null;

    var signatureUrl = Request.Headers.Contains("SignatureCertChainUrl")
        ? Request.Headers.GetValues("SignatureCertChainUrl").FirstOrDefault()
        : null;

    if (signature == null || signatureUrl == null)
        return false;

    var requestStr = await Request.Content.ReadAsStringAsync();
    var alexaRequestBytes = Encoding.UTF8.GetBytes(requestStr);
    return await AlexaSkillsKit.Authentication.SpeechletRequestSignatureVerifier.VerifyRequestSignatureAsync(
            alexaRequestBytes, signature, signatureUrl);
}
``` 
### Issue
My problem is, this last call to `VerifyRequestSignatureAsync` **always returns `false`**, so no request can get processed and my API returns `400 BadRequest`. How can I get this work, **is there any initialisation, or configuration of the package that I'm missing?**

I'm using AlexaSkillsKit.NET v1.6.0 with its dependency BouncyCastle v1.8.1.

### More info
To understand exactly where this fails, I copied the implementation of this method inside my controller and added logging between each step and found out that the last step `signer.VerifySignature(expectedSignatureBytes);` is that always fails, all the previous steps about certificate validation and url validation are validated successfully.

Just for reference, here is the part I copied (last line always returns false):

```
var url = new Uri(signatureUrl);
var correctScheme = url.Scheme.ToLower() == "https";
var correctHost = url.Host.ToLower() == "s3.amazonaws.com";
var correctPath = url.PathAndQuery.StartsWith("/echo.api/");
var correctPort = url.IsDefaultPort;
var isValidUrl = correctScheme && correctHost && correctPath && correctPort;

if (!isValidUrl) return false;

using (var httpClient = new HttpClient()) {
    var response = await httpClient.GetAsync(url);
    if (!response.IsSuccessStatusCode) return false;
    var body = await response.Content.ReadAsStringAsync();
    var pemReader = new Org.BouncyCastle.OpenSsl.PemReader(new StringReader(body));
    var cert = (X509Certificate) pemReader.ReadObject();
    try {
        cert.CheckValidity();
        if (!CheckCertSubjectNames(cert)) {
            _logger.Debug("Validation - CheckCertSubjectNames = false");
            return false;
        }
    }
    catch (CertificateExpiredException e) {
        _logger.Error("Validation - cert.CheckValidity throws CertificateExpiredException", e);
        return false;
    }
    catch (CertificateNotYetValidException e) {
        _logger.Error("Validation - cert.CheckValidity throws CertificateNotYetValidException", e);
        return false;
    }

    byte[] expectedSignatureBytes;
    try {
        expectedSignatureBytes = Convert.FromBase64String(signature);
    }
    catch (FormatException e) {
        _logger.Error("Validation - expectedSignatureBytes = Convert.FromBase64String(expectedSignature)",
            e);
        return false;
    }

    var publicKey = (Org.BouncyCastle.Crypto.Parameters.RsaKeyParameters) cert.GetPublicKey();
    var signer = Org.BouncyCastle.Security.SignerUtilities.GetSigner("SHA1withRSA");
    signer.Init(false, publicKey);
    signer.BlockUpdate(alexaRequestBytes, 0, alexaRequestBytes.Length);

    _logger.Debug("Validation - final verification");
    return signer.VerifySignature(expectedSignatureBytes);
}
```

**Comment by zafeirism** [comment]
I found the problem, which was not related to this library. Line

`var requestStr = await Request.Content.ReadAsStringAsync();`

always returns an empty string because my controller endpoint were using model binding (more about this [here](http://www.hackered.co.uk/articles/asp-net-web-api-why-is-the-request-Content-empty-when-the-model-is-populated)). When I fixed this, validation works as expected.
